### PR TITLE
add telnyx and plivo

### PR DESCRIPTION
**Pull request by benlower**
also added new mediums to create call

### CODE REVIEW

File: docs/src/content/docs/guides/connectionoptions.mdx [hunk 1]
```diff
@@ -54,28 +49,77 @@ Creating an Ultravox call that works with Twilio is just like creating a WebRTC
...
+    });
+    ```
+
+    Or using TeXML:
+    ```xml
+    <?xml version="1.0" encoding="UTF-8"?>
+    <Response>
+      <Start>
+        <Stream url="${joinUrl}" track="both_tracks" />
+      </Start>
```
Comment (mdepinet) [inline docs/src/content/docs/guides/connectionoptions.mdx:92]: I think this should be `<Connect><Stream url=${your-join-url} /></Connect>` but I still can't test this.  Given the issues with my account, maybe we can have Zach invite you instead and you can test it out.

The JS above seems reasonable, but I don't have a way to verify that either.

File: docs/src/content/docs/guides/connectionoptions.mdx [hunk 2]
```diff
@@ -54,28 +49,77 @@ Creating an Ultravox call that works with Twilio is just like creating a WebRTC
...
+#### Incoming Calls with Telnyx
+<Steps>
+1. **Create an Ultravox Call** → Create a new call with `medium: { "telnyx": {} }` and `firstSpeaker` set to "FIRST_SPEAKER_AGENT".
+
+2. **Handle Inbound Call** → Use the `joinUrl` with a TeXML `<Stream>`:
+
+    ```xml
+    <?xml version="1.0" encoding="UTF-8"?>
+    <Response>
+      <Start>
```
Comment (mdepinet) [inline docs/src/content/docs/guides/connectionoptions.mdx:106]: Same as above.

File: docs/src/content/docs/guides/connectionoptions.mdx [hunk 3]
```diff
@@ -85,37 +129,92 @@ It only takes two steps to make an outgoing call to regular phone numbers throug
...
+    ```
+
+    The answer URL should return:
+    ```xml
+    <?xml version="1.0" encoding="UTF-8"?>
+    <Response>
+        <Stream url="${joinUrl}" 
+                audioTrack="both" 
+                contentType="audio/x-l16;rate=8000" 
+                bidirectional="true" />
```
Comment (mdepinet) [inline docs/src/content/docs/guides/connectionoptions.mdx:182]:
```suggestion
        <Stream keepCallAlive="true" 
                contentType="audio/x-l16;rate=16000"
                bidirectional="true">
            ${joinUrl}
        </Stream>
```

Maybe also worth noting that any contentType supported by Plivo will be fine for Ultravox, but we recommend audio/x-l16;rate=16000 for the best audio quality.

File: docs/src/content/docs/guides/connectionoptions.mdx [hunk 4]
```diff
@@ -85,37 +129,92 @@ It only takes two steps to make an outgoing call to regular phone numbers throug
...
+1. **Create an Ultravox Call** → Create a new call as shown above with `medium: { "plivo": {} }`, `firstSpeaker: "FIRST_SPEAKER_USER"`, and get a `joinUrl`.
+
+2. **Initiate Plivo Phone call** → Use the `joinUrl` with AudioStream:
+
+    ```javascript
+    // Example using the plivo node library
+    const call = await plivo.calls.create({
+      to: phoneNumber,
+      from: plivoPhoneNumber,
+      answer_url: answerUrl, // URL that returns the XML below
```
Comment (mdepinet) [inline docs/src/content/docs/guides/connectionoptions.mdx:170]: nit: This is actually true for all the providers, but given that Plivo doesn't have any way to use the joinUrl here directly, I'd either omit this bit or change things up to make it more clear that your server should have an endpoint (used here) that creates a call whose joinUrl is used in the XML response.

File: docs/src/content/docs/guides/connectionoptions.mdx [hunk 5]
```diff
@@ -85,37 +129,92 @@ It only takes two steps to make an outgoing call to regular phone numbers throug
...
+
+2. **Handle Inbound Call** → Use the `joinUrl` with AudioStream:
 
+    ```xml
+    <?xml version="1.0" encoding="UTF-8"?>
+    <Response>
+        <Stream url="${joinUrl}" 
+                audioTrack="both" 
+                contentType="audio/x-l16;rate=8000" 
+                bidirectional="true" />
```
Comment (mdepinet) [inline docs/src/content/docs/guides/connectionoptions.mdx:199]:
```suggestion
        <Stream keepCallAlive="true"
                contentType="audio/x-l16;rate=16000" 
                bidirectional="true">
            ${joinUrl}
        </Stream>
```

File: docs/src/content/docs/guides/connectionoptions.mdx [hunk 6]
```diff
@@ -85,37 +129,92 @@ It only takes two steps to make an outgoing call to regular phone numbers throug
...
+</TabItem>
+</Tabs>
+
+### Best Practices
+1. Always test your integration thoroughly in a development environment before deploying to production
+2. Implement proper error handling and fallback logic
+3. Monitor call quality and connection stability
+4. Ensure your system can handle the audio stream bidirectionally
+5. Consider implementing reconnection logic for dropped connections
+
```
Comment (mdepinet) [inline docs/src/content/docs/guides/connectionoptions.mdx:214]: I assume these are also generated by Claude?  I don't think they belong here as they're either overly generic, insufficiently helpful, or just plain misleading.
```suggestion
```

File: docs/src/content/docs/api/calls.mdx [hunk 7]
```diff
@@ -184,7 +184,7 @@ An optional query parameter called `priorCallId` can be provided to continue a p
       <tr>
         <td class="font-mono">medium</td>
         <td>object</td>
-        <td>Details about a call's protocol. By default, calls occur over WebRTC using the Ultravox client SDK. Setting a different call medium will prepare the server for a call using a different protocol. At most one call medium may be set.<br />Can be `"webRtc":{}` (default) or `"twilio":{}`.</td>
+        <td>Details about a call's protocol. By default, calls occur over WebRTC using the Ultravox client SDK. Setting a different call medium will prepare the server for a call using a different protocol. At most one call medium may be set.<br />Can be `"webRtc":{}` (default), `"telnyx":{}`, `"twilio":{}`, `"plivo":{}`, or `"serverWebSocket":{}`.</td>
```
[the remark this answers is not in the document]
Comment (benlower) [reply docs/src/content/docs/api/calls.mdx:187]: yeah linking is better. done.

File: docs/src/content/docs/guides/connectionoptions.mdx (hunk 4 above)
Comment (benlower) [reply docs/src/content/docs/guides/connectionoptions.mdx:170]: added more content to clarify

File: docs/src/content/docs/guides/connectionoptions.mdx (hunk 1 above)
Comment (benlower) [reply docs/src/content/docs/guides/connectionoptions.mdx:92]: ok making this change for now. will fix up if needed once we can test fully.

File: docs/src/content/docs/guides/connectionoptions.mdx (hunk 2 above)
Comment (benlower) [reply docs/src/content/docs/guides/connectionoptions.mdx:106]: yes

**Pull request merged**